model_filereading.cpp

```diff
@@ -10,19 +10,40 @@
 #include <iterator>
 
 // split line into substrings when separated by whitespaces
-static inline std::vector<std::string> splitLine(std::string& line){ 
+static inline std::vector<std::string> splitLine(std::string& line){
   std::istringstream iss(line);
   std::vector<std::string> substrings((std::istream_iterator<std::string>(iss)), std::istream_iterator<std::string>());
   return substrings;
 }
 
-// reads radii from a file specified by the filepath and 
+// Reads a string and converts it to valid atom symbol with first char upper case then lower case chars
+std::string strToValidSymbol(const std::string& str_inp){
+  std::string validSymbol = "";
+  for (int i = 0; str_inp[i]!=0; i++) {
+    if (isalpha(str_inp[i])) {
+      validSymbol = validSymbol + str_inp[i];
+      if (i==0 && islower(validSymbol[i])) {
+        validSymbol[i] = toupper(validSymbol[i]);
+      }
+      else if (i>0 && isupper(validSymbol[i])) {
+        validSymbol[i] = tolower(validSymbol[i]);
+      }
+    }
+    else {
+      return validSymbol;
+    }
+  }
+}
+
+// reads radii from a file specified by the filepath and
```

Comment:
A few things:

The function can theoretically reach the end without a return value. If you use the -Werror flag when compiling, this should come up as an error, so I wonder how this compiled for you.

`toupper()` converts single characters to uppercase, but only if possible. Otherwise it returns the input. That means it is redundant to check `islower(c)` before calling `toupper(c)`. Same for `tolower()`.

It is not necessary to create an output variable here. If you change the argument from `f(const type& arg)` to `f(type arg)`, then a copy of the input variable is created when this function is called. You can then manipulate the input variable and use it as return variable.

Finally, stylistically I prefer to use camelCase for functions and snake_case for variables. Imo it makes distinguishing between the two much easier. This means I'd have used `valid_symbol` instead.